apps/server/src/shared/common/validator/date.validator.ts

```diff
@@ -0,0 +1,12 @@
+import { ValidatorConstraint, ValidatorConstraintInterface, ValidationArguments } from 'class-validator';
+
+@ValidatorConstraint({ name: 'isBefore', async: false })
+export class IsBeforeConstraint implements ValidatorConstraintInterface {
+	validate(propertyValue: Date, args: ValidationArguments) {
+		return propertyValue < args.object[args.constraints[0] as unknown as string];
```

Comment:
this "as..." structure looks really weird to me to be honest, but if you say thats the only way ^^